File: pset6/readability.py
from cs50 import get_string

n = get_string("Text: ")

letters = words = sentences = 0

for char in n:
    if char.isalpha():
        letters += 1
    if char.isspace():
        words += 1
    if char in ["?", "!","."]:
        sentences += 1

words += 1

L = (letters * 100.00) / words
S = (sentences * 100.00) / words

index = round(0.0588 * L - 0.296 * S - 15.8)

if index >= 16:
    print("Grade 16+")
elif index <= 1:
    print("Before Grade 1")
else:
    print(f"Grade {index}")




#Coleman Liu Index = 0.0588 * L - 0.296 * S - 15.8
# Letters for 100 words and Sentences for 100 words

File: pset1/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int main(void)

{   // Prompt for user input string
    string text = get_string("Text:\n");

    //find out number of words, sentences, letters in a text
    int letters = 0;

    int sentences = 0;

    int words = 0;

    // Letters count, accurate
    for (int i = 0; text[i]!='\0'; i++ )

            if (isalpha( (unsigned char) text [i] ) )
            {
                letters++;
             }

    // Sentences Count,accurate
    for (int j = 0; j < strlen (text); j++ )

            if (text[j] == '!' || text[j] == '?' || text[j] == '.')
            {
                sentences++;
            }

    // words count. Accuracy = Less 1 than actual. Fixed in next line
   for (int k = 0; text[k]!='\0'; k++ )

            if (isblank( (unsigned char) text [k] ) )
            {
                words++;
            }

    //fixing word count to account for last full stop.
    words = words + 1;

    float a = letters * 100.00 / words;

    float b = sentences * 100.00 / words;

    float index = 0.0588 * a - 0.296 * b - 15.8;

    int clindex = round(index);

    if (clindex >= 16 )
    {
        printf("Grade 16+\n");
    }
    else if (clindex < 1 )
    {
        printf("Before Grade 1\n");
    }
    else  
    {  
        printf("Grade %i\n", clindex);
    }

}
